packages/web/src/components/product/FieldBadge.tsx

```
@@ -0,0 +1,245 @@
+export function FieldBadge({ provenance, fieldPath, className = '' }: FieldBadgeProps) {
+  const [showTooltip, setShowTooltip] = useState(false);
+  const badgeRef = useRef<HTMLButtonElement>(null);
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  // Don't render if no provenance or not from Smart Rule
+  if (!provenance || provenance.source !== 'smartRule') {
+    return null;
+  }
+
+  // Close tooltip on outside click
+  useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)